src/drivers/actuators/vertiq_io/vertiq_client_manager.hpp

```diff
@@ -0,0 +1,225 @@
+	*/
+	VertiqClientManager(VertiqSerialInterface *serial_interface);
+
+	/**
+	* @brief Initialize all of our clients with the object ID given by the PX4 parameter TARGET_MODULE_ID
+	*/
+	void Init(uint8_t object_id);
+
+	/**
+	* @brief Handle the IQUART interface. Make sure that we update TX and RX buffers
+	*/
+	void HandleClientCommunication();
+
+	/**
+	* @brief Returns access to our IFCI interface
+	* @return A pointer to our IFCI interface _motor_interface
+	*/
+	IFCI *GetMotorInterface();
+
+	/**
+	* @brief Add a set to the output buffer that will force the connected motor to arm
+	*/
+	void SendSetForceArm();
+
+	/**
+	* @brief Add a set to the output buffer that will force the connected motor to disarm
+	*/
+	void SendSetForceDisarm();
+
+	/**
+	* @brief Add a set to the output buffer that will force the connected motor to coast
+	*/
+	void SendSetCoast();
+
+	/**
+	* @brief Add a set to the output buffer that will force the connected motor to spin at a given setpoint
+	* @param velocity_setpoint the raw 16-bit velocity command going to the motor
+	*/
+	void SendSetVelocitySetpoint(uint16_t velocity_setpoint);
+
+	/**
+	* @brief Returns the number of clients that we've initialized
+	* @return The value contained in _clients_in_use
+	*/
+	uint8_t GetNumberOfClients();
+
+	/**
+	* @brief Set all of the IQUART configuration init flags to true
+	*/
+	void MarkIquartConfigsForRefresh();
+
+	/**
+	* @brief Send a Get command to all of the parameters involved in IQUART configuration, and make sure the PX4 parameters and module values agree
+	*/
+	void UpdateIquartConfigParams();
+
+	/**
+	* @brief Until the timeout is reached, keep trying to update the PX4 parameters to match, as appropraite, the value on the module. This can
+	*	mean either setting the PX4 parameter to match the motor or vice versa
+	*/
+	void CoordinateIquartWithPx4Params(hrt_abstime timeout = 2_s);
+
+	/**
+	* @brief Gives access to the object ID currently being used
+	* @return The value stored in _object_id_now
+	*/
+	uint8_t GetObjectIdNow();
+
+	/**
+	* @brief When the target module ID parameter changes, we need to delete and remake all of our clients in order to make sure that we're talking to the
+	*        correct motor.
+	* @param new_object_id The new target module ID that we should use to instantiate our new clients
+	*/
+	void UpdateClientsToNewObjId(uint8_t new_object_id);
+private:
+	/**
+	* @brief Initialize all of the Vertiq Clients that we want to use
+	* @param object_id The object ID with which to initialize our clients
+	*/
+	void InitVertiqClients(uint8_t object_id);
+
+	/**
+	* @brief Initialize all of the combo_entry structs
+	*/
+	void InitComboEntries();
+
+	uint8_t _object_id_now;
+
+	//We need a serial handler in order to talk over the serial port
+	VertiqSerialInterface *_serial_interface;
+
+	//IQUART Client configuration
+	IFCI _motor_interface;
+
+	//These exist whenever we're using IQUART
+	EntryWrapper<float, float> _velocity_max_entry;
```

Comment:
here's where we make all of the EntryWrappers...duh. If a user wanted to add a new one, they'd have to do it here.